core-bundle/src/Framework/ContaoFramework.php

```diff
@@ -427,10 +427,12 @@ private function setTimezone()
      */
     private function triggerInitializeSystemHook()
     {
-        if (isset($GLOBALS['TL_HOOKS']['initializeSystem']) && \is_array($GLOBALS['TL_HOOKS']['initializeSystem'])) {
-            foreach ($GLOBALS['TL_HOOKS']['initializeSystem'] as $callback) {
-                System::importStatic($callback[0])->{$callback[1]}();
-            }
+        if (empty($GLOBALS['TL_HOOKS']['initializeSystem']) || !is_dir($this->rootDir.'/system/tmp')) {
+            return;
```

Comment:
This would always return when no `initializeSystem` hook is present and thus the `initconfig.php` will never be loaded in that case for example.

Reply:
That's intended as discussed in the last Mumble call (see https://github.com/contao/contao/issues/929#issuecomment-611531123). We do not want to execute any logic unless the system is fully installed.

Reply:
@ausi @aschempp /cc

Reply:
Yes, but this way the `initconfig.php` is _never_ executed, regardless of whether the system is fully installed or not, when no hook is registered.

Reply:
To be clear: I mean
```php
if (empty($GLOBALS['TL_HOOKS']['initializeSystem']) || …) {
    return;
}
```

If no `intializeSystem` hooks is registered, the function returns early, meaning that the `initconfig.php` for example, which would be included later on, is never included.

Reply:
You are absolutely right. My bad. 🤦‍♂️

Reply:
Changed in f1de5eb44130701bdefb63e6f37aefdf43d17de6.

Reply:
LGTM.